test/unit-test/cellular_at_core_utest.c

```diff
@@ -0,0 +1,932 @@
+    cellularStatus = Cellular_ATGetSpecificNextTok( & pString, pDelimiter, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATGetSpecificNextTok to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATGetSpecificNextTok_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    const char * pDelimiter = ",";
+    char * pTokOutput;
+    char * pString = malloc( sizeof( char ) * ( strlen( CELLULAR_SAMPLE_STRING_TWO_DOUBLE_QUOTE ) + 1 ) );
+    char * pStringSource = pString;
+    
+    pString = malloc( sizeof( char ) * ( strlen( CELLULAR_SAMPLE_STRING_DELIMTER_FIRST ) + 1 ) );
+    pStringSource = pString;
+    strcpy( pString, CELLULAR_SAMPLE_STRING_DELIMTER_FIRST );
+    /* Check that CELLULAR_AT_SUCCESS is returned. */
+    cellularStatus = Cellular_ATGetSpecificNextTok( & pString, pDelimiter, & pTokOutput );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+    free( pStringSource );
+
+    pString = malloc( sizeof( char ) * ( strlen( CELLULAR_SAMPLE_WRONG_STRING_DELIMTER_FIRST ) + 1 ) );
+    pStringSource = pString;
+    strcpy( pString, CELLULAR_SAMPLE_WRONG_STRING_DELIMTER_FIRST );
+    /* Check that CELLULAR_AT_SUCCESS is returned. */
+    cellularStatus = Cellular_ATGetSpecificNextTok( & pString, pDelimiter, & pTokOutput );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+    free( pStringSource );
+}
+
+/**
+ * @brief Test the error path for Cellular_ATGetSpecificNextTok to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATGetSpecificNextTok_Error_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString = "";
+    const char * pDelimiter = ",";
+    char * pTokOutput;
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATGetSpecificNextTok( & pString, pDelimiter, & pTokOutput );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test that any NULL parameter causes Cellular_ATHexStrToHex to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATHexStrToHex_Invalid_Param( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString;
+    uint8_t * const * const ppHexData;
+    uint16_t hexDataLen;
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATHexStrToHex( NULL, ppHexData, hexDataLen );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATHexStrToHex( pString, NULL, hexDataLen );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATHexStrToHex to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATHexStrToHex_Happy_Path( void )

```

Comment:
This can be broken in two too:
* test_Cellular_ATHexStrToHex_Happy_Path_Uppercase_Hex_String
* test_Cellular_ATHexStrToHex_Happy_Path_Lowecase_Hex_String